controllers/dscinitialization/dscinitialization_test.go

```diff
@@ -0,0 +1,294 @@
+package dscinitialization
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	dsci "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	authv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Dscinitialization", func() {
+	Context("Test DSCInitialization controller", func() {
+		It("Test if all the resources are created", func() {
+			By("By creating a new DSC instance")
+			ctx := context.Background()
+			dsci := &dsci.DSCInitialization{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "DSCInitialization",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      AppName,
+					Namespace: WorkingNamespace,
+				},
+				Spec: dsci.DSCInitializationSpec{
+					ApplicationsNamespace: AppNamespace,
+					Monitoring: dsci.Monitoring{
+						Namespace:       MonitoringNamespace,
+						ManagementState: operatorv1.Managed,
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, dsci)).Should(Succeed())
+			// wait for a few seconds for the cluster to spin up.
+			time.Sleep(time.Second * 5)
```

Comment:
Depending on the environment it might take way less time so sleeping 5s adds extra cost for running tests or might fail if environment is slower. Perhaps it would be better to wait with a timeout for a condition instead?